aviary/mission/gasp_based/ode/time_integration_base_classes.py

```diff
@@ -729,7 +811,7 @@ def compute_partials(self, inputs, J):
 
                             dh_j_dx = prob.compute_totals(
                                 [state_name],
-                                prob.states.keys(),
+                                list(prob.states.keys()),
```

Comment:
You should probably just turn these into a list once and save it to a local var. This is several loops deeper from the loop over probs.